HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,128 @@
+
+
+### Graph 1
+```{r, echo = T, include = TRUE}
+data("diamonds")
+#hint think about the *position* the bars are in...
+```
+
+
+Using the diamonds dataset, make this graph:
+```{r graph1 code, echo=FALSE}
+ggplot(diamonds, aes(cut, fill=clarity)) + 
+  geom_bar(position = "dodge") + 
+  labs(title="My Diamond Collection", 
+        subtitle="Boxplot representing the number of diamonds in my diamond collection          by \n type of cut quality and clarity of diamond", 
+        x = "Diamond Cut", y = "Number of Diamonds") + 
+  theme(plot.title = element_text(hjust = 0.5), 
+        plot.subtitle = element_text(size=9)) +
+  annotate(geom = "text", x = "Premium", y = 4500, 
+        label = "My Best Diamonds, \n of course") + 
+  annotate(geom = "rect", xmin=4.5, xmax=5.5, ymin=0, ymax= 5000, alpha=.3)
+
+
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+```
+
+Using the iris dataset, make this graph:
+
+**I figured out facet_wrap, but I am certainly not going to rearange the entire dataframe just to get setosa and versicolor to swtich spots. I wont do it!!!**
+
+```{r graph 2 code, echo=FALSE}
+
+ggplot(iris, aes(Sepal.Length, Petal.Length, color=Species, shape=Species)) +
+  geom_point() + 
+  geom_smooth(method = "lm", se=F, color = "black") +
+  facet_wrap(vars(Species), scales = "free_y")
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graoh 3 code}
+label <- paste("Corvette" ,corvette$year, sep=", ")
+
+ggplot(mpg, aes(displ, hwy)) +
+  geom_point(size=.75) +
+  geom_point(data=corvette, aes(displ, hwy), color ="blue", size=.75)+
+  geom_text_repel(data=corvette, aes(label=label), size = 3) +
+  labs(subtitle="Corvettes are a bit of an outlier")+
+  theme(plot.subtitle = element_text(size=12))
+```
+
+There is a trick to getting the model and year to print off together. `paste()` is a useful function for this, also pasting together parts of file names and parts of urls together. 
+
+
+### Graph 4 
+```{r, echo = T}
+data(mpg)
+
+#hint for the coloring, colorbrewer and you can set palette colors and make your graphs colorblind friendly
+library(RColorBrewer)
+display.brewer.all(colorblindFriendly = T) #take a look at the colorblindfriendly options
+```
+
+The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
+
+
+
+Now using the above mpg dataset, make this graph
+
+```{r graph 4 code}
+ggplot(mpg, aes(class, cty))+
+  geom_boxplot(outlier.shape = NA) + coord_flip()+
+  scale_color_brewer(palette="Set2")+
+  geom_jitter(aes(color=class),width=.5, height=0, size=.7)+
+  labs(subtitle="Horizontal Boxplot of City MPG and Car Class", 
+      x= "City MPG", y= "Car Class")
+```
+
```

Comment:
Great job with this difficult plot. I struggled with that one the most.
Some things you could add: remove the border of the plot and the grid lines, and add the axes. You could do that using:

theme(
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), 
    panel.border = element_blank(),
    axis.line = element_line(colour = "black")
    )

Reply:
Ah I think I only noticed that difference in the graph right at the end and decided not to look into it. Thank you!